tap_pipedrive/tap.py

```diff
@@ -244,17 +244,43 @@ def execute_stream_request(self, stream):
         return self.execute_request(stream.endpoint, params=params)
 
     def execute_request(self, endpoint, params=None):
-        headers = {"User-Agent": self.config["user-agent"]}
-        _params = {
-            "api_token": self.config["api_token"],
-        }
-        if params:
-            _params.update(params)
-
         url = "{}/{}".format(BASE_URL, endpoint)
-        logger.debug("Firing request at {} with params: {}".format(url, _params))
+        logger.debug("Firing request at {} with params: {}".format(url, params))
 
-        return requests.get(url, headers=headers, params=_params)
+        # remove this case when all customers have authed with oauth
+        if "api_token" in self.config:
+            params = {"api_token": self.config["api_token"], **params}
+            headers = {"User-Agent": self.config["user-agent"]}
+            response = requests.get(url, headers=headers, params=params)
```

Comment:
I'd just add a return here and eliminate the nested else below:

```python
return requests.get(url, headers=headers, params=params)
```

Reply:
👍